common/djangoapps/pipeline_mako/__init__.py

```diff
@@ -79,3 +82,78 @@ def render_individual_js(package, paths, templates=None):
     if templates:
         tags.append(render_inline_js(package, templates))
     return '\n'.join(tags)
+
+
+def render_require_js_path_overrides(path_overrides):  # pylint: disable=invalid-name
+    """Render JavaScript to override default RequireJS paths.
+
+    The Django pipeline appends a hash to JavaScript files,
+    so if the JS asset isn't included in the bundle for the page,
+    we need to tell RequireJS where to look.
+
+    For example:
+
+        "js/vendor/jquery.min.js" --> "js/vendor/jquery.min.abcd1234"
+
+    We would then add a line in a <script> tag:
+
+        require.paths['jquery'] = 'js/vendor/jquery.min.abcd1234'
+
+    so that any reference to 'jquery' in a JavaScript module
+    will cause RequireJS to load '/static/js/vendor/jquery.min.abcd1234.js'
+
+    If running in DEBUG mode (as in devstack), the resolved JavaScript URLs
+    won't contain hashes, so the new paths will match the original paths.
+
+    Arguments:
+        path_overrides (dict): Mapping of RequireJS module names to
+            filesystem paths.
+
+    Returns:
+        unicode: The HTML of the <script> tag with the path overrides.
+
+    """
+    # Render the <script> tag that overrides the paths defined in `require.paths`
+    # Note: We don't use a Mako template to render this because Mako apparently
+    # acquires a lock when loading templates, which can lead to a deadlock if
+    # this function is called from within another template.
+    html = ['<script type="text/javascript">']
+
+    # The rendered <script> tag with overrides should be included *after*
+    # the application's RequireJS config, which defines a `require` object.
+    # Just in case the `require` object hasn't been loaded, we create a default
+    # object.  This will avoid a JavaScript error that might cause the rest of the
+    # page to fail; however, it may mean that these overrides won't be available
+    # to RequireJS.
+    html.extend([
+        'var require = require || {};',
```

Comment:
I think you need to use ``RequireJS.require`` when this code is executed in the LMS. The LMS uses a namespace for RequireJS so that it doesn't perturb JavaScript files that behave differently in its presence. See this wiki for details: https://openedx.atlassian.net/wiki/display/TNL/RequireJS+in+the+LMS

This code probably needs to be:

    `var require = require || RequireJS.require;',

You don't ever want to default to ``{}`` because then the configuration will be added to an object that is never referenced.